meta-hoki/recipes-core/initrdscripts/initramfs-scripts-android/init.machine.sh

```diff
@@ -17,4 +17,24 @@ if [ ! -e $BOOT_DIR/etc/asteroid/machine.conf ] ; then
     echo Fossil Gen 6 > /sys/kernel/config/usb_gadget/g1/strings/0x409/product
 
     /usr/bin/adbd
+}
+
+# wait for the middle button to be pressed, and if it is,
+# kill the calling process group with SIGUSR1
+waitmidbutton() {
+    fmtstring='1/4 "%d." 1/4 "%d\t" 1/2 "%2x\t" 1/2 "%2x\t" 1/4 "%u\n"'
```

Comment:
This looks a bit cryptic, perhaps add a description of how the bytes are parsed?

Leaving it up to you. Feel free to merge regardless.

Reply:
I was wondering if this was common knowledge or not.  I'll document better.